kroxylicious/pom.xml

```diff
@@ -263,5 +263,39 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>architecture-check</id>
+            <repositories>
+                <repository>
+                    <id>jitpack</id>
+                    <name>Jitpack</name>
+                    <url>https://jitpack.io</url>
+                </repository>
+            </repositories>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-compiler-plugin</artifactId>
+                        <configuration>
+                            <fork>true</fork>
+                            <showWarnings>true</showWarnings>
+                            <compilerArgs>
+                                <!-- <arg>-Xplugin:Deptective mode=ANALYZE components=io.netty.*:io.netty*;org.apache.kafka.*:org.apache.kafka*;io.kroxylicious.proxy.example.*:io.kroxylicious.proxy.example* visualize=true</arg> -->
+                                <arg>-Xplugin:Deptective mode=VALIDATE cycle_reporting_policy=ERROR reporting_policy=WARN visualize=true</arg>
```

Comment:
So I guess all I really want to know is:

* Will this fail the CI when the rules are broken?
* Will the report contain the cute graph visualisation?